src/jarabe/model/shell.py

```diff
@@ -668,8 +668,7 @@ def _active_window_changed_cb(self, screen, previous_window=None):
                 window = window.get_transient()
 
         act = self._get_activity_by_xid(window.get_xid())
-        if act is not None:
-            self._set_active_activity(act)
+        self._set_active_activity(act)
```

Comment:
All right @samdroid-apps I was able to reproduce but doesn't always happen. We'll figure it out.